src/cr/cube/matrix.py

```diff
@@ -2527,7 +2532,51 @@ def values(self):
         return self._counts
 
 
-class _CatXMrVector(_CategoricalVector):
+class _MeansVector(_BaseVector):
+    """Used on a CAT dimension of a CAT_X_CAT matrix with means cube-result measure."""
+
+    def __init__(self, element, unweighted_counts, means):
+        super(_MeansVector, self).__init__(element, unweighted_counts)
+        self._unweighted_counts = unweighted_counts
+        self._means = means
+        self._opposing_margin = None
+
+    @lazyproperty
+    def means(self):
+        """1D np.float64/np.nan ndarray of mean for each vector cell.
+
+        Mean is np.nan for a cell with an unweighted-count of zero.
+        """
+        return self._means
+
+    @lazyproperty
+    def values(self):
+        return self._means
+
+    @lazyproperty
+    def margin(self):
+        """Unconditionally np.nan for a means vector.
+
+        A means vector has no (weighted) counts, so it can have no margin.
+        """
+        return np.nan * len(self._unweighted_counts)
+
+
+class _MeansWithMrVector(_MeansVector):
+    """Used for row vectors in a means matrix with an MR dimension."""
+
+    # TODO: Work out why non-means vectors are used for the columns of these matrices.
+    # I expect it is because _Slice gets most of its values from rows and columns are
+    # only used for column-oriented things like column-proportions and column-labels.
+    # Still, could be bugs lurking here as to how columns handle means.
+
+    @lazyproperty
+    def base(self):
+        """np.int64 unweighted-N for this vector."""
+        return np.sum(self._unweighted_counts[0])
+
+
+class _MrOpposingCatVector(_CategoricalVector):
```

Comment:
I do like this name better than the old one. Although the old one also made some sense (at least to me when I was designing it). I don't remember what that sense was though, at this time.

Anyways, I think we can be even more precise with this name. This is good for now, but maybe something more mathematically sound would be better. What is this vector actually? Is it an "row of a mr x cat matrix" most of the time? Or is it a "column of a cat x mr matrix"? Or both?

The word that's currently ringing in my mind is "orthogonal", potentially giving a name similar to `MrOtrhogonalToCatVector`, but this might be too cumbersome (if even true).

Leave it to your best judgement, but this is also good for now.

Reply:
It's used for rows of MR_X_CAT and for columns of CAT_X_MR. It's an abstraction of that symmetry, like if you're an MR vector and your opposing vectors are CAT then this is the vector class for you.

I discussed preferring "orthogonal" to "opposing" with Mike and he said he knew of no statistical-world convention of using "orthogonal" in this way that we might want to respect as a domain convention, and also that he found "opposing" descriptive. Add that to the fact that we use it broadly now and I'm not going to change it at the moment. I will keep it in the back of my mind though and continue to consider it.